Einführung der Authentifizierungs-Middleware und Verbesserungen der Projektkonfiguration

calipee opened the pull request:
Dieser Pull Request umfasst folgende Änderungen:

1. **Erstellen einer Authentifizierungs-Middleware für Basic Authentication**
   - Implementierung einer Middleware, die die grundlegende Authentifizierung unterstützt, um den Zugriff auf bestimmte Ressourcen zu sichern.

2. **Hinzufügen der Middleware-Konfiguration für passende Routen**
   - Konfiguration der Middleware, sodass sie nur auf ausgewählte Routen angewendet wird, was Flexibilität und Sicherheit erhöht.

3. **Einführung von `.env.example` für die Einrichtung von Umgebungsvariablen**
   - Hinzufügen einer Beispiel-Umgebungsvariablen-Datei, um Entwicklern den Einstieg und die korrekte Konfiguration zu erleichtern.

4. **Aktualisierung der `.gitignore`, um das `.vscode`-Verzeichnis auszuschließen**
   - Anpassung der `.gitignore`, um persönliche Einstellungen des Visual Studio Code Editors nicht mit ins Repository aufzunehmen.

Bitte überprüft die Änderungen und merged sie bei Zustimmung.

calipee committed:
Add basic authentication middleware and example environment file

- Create auth middleware for basic authentication
- Add middleware configuration for matching routes
- Introduce .env.example for environment variable setup
- Update .gitignore to exclude .vscode directory

DanielEdlich commented on src/middleware/auth.ts:
```diff
@@ -0,0 +1,25 @@
+import { NextResponse } from "next/server";
+import type { NextRequest } from "next/server";
+
+const username = process.env.BASIC_AUTH_USERNAME || "admin";
```
no default username and password

calipee replied:
Entfernt und wirft jetzt Fehler wenn nicht gesetzt

DanielEdlich commented on src/middleware/auth.ts:
```diff
@@ -0,0 +1,25 @@
+const username = process.env.BASIC_AUTH_USERNAME || "admin";
+const password = process.env.BASIC_AUTH_PASSWORD || "passwort";
+
+export function authMiddleware(request: NextRequest) {
```
what about testing?

calipee replied:
Habe Tests hinzugefügt

DanielEdlich commented on src/middleware.ts:
```diff
@@ -0,0 +1,12 @@
+import type { NextRequest } from "next/server";
+import { authMiddleware } from "./middleware/auth";
+
+export function middleware(request: NextRequest) {
```
what about testing?

calipee replied:
Habe Tests hinzugefügt

DanielEdlich commented on .env.example:
can you create  a env that defines a login and password for local development and also loads this env in dev mode

calipee replied:
Ich weiß nicht genau was du damit meinst? Für die lokale Entwicklung solltest du dir ein `.env.local` File erstellen in dem du wie in `.env.example` die Variablen einträgst. Dieses sollte allerdings nicht committed werden.

RemodifyBerlin replied:
Felix Lösung passt. env.local sollte nicht in repo sein. Es hätte auch gereicht, es in der readme kurz zu beschreiben

calipee replied:
Habe in README dokumentiert das eine `.env` Datei erzeugt werden sollte

calipee committed:
SCRUM-18 Enhance authentication middleware and add tests

- Update package.json to include ts-jest and upgrade typescript version
- Configure Jest to use ts-jest preset
- Improve error handling in auth middleware for missing credentials
- Update README to include instructions for creating a .env file
- Add unit tests for middleware and authMiddleware functionality
